### PR TITLE
broker: move policy config check out to the modules that rely on it

### PR DESCRIPTION
**Pull request by garlick**
Problem: code was added to the broker, originally by a7760fc52cef445e886d73110698335289680f8d, to check RFC 33 policy/queue configuration because it was thought awkward to check it elsewhere.  Re-evaluating it now, it seems like a misstep.  It's probably better if the broker remains unaware of such details in order to maintain a separation of concerns.

Anyway it's easy to fix.  Move checks into job-ingest and job-manager (with shared code moved to a library).

Also drop an embarrassing chunk of dead code that was accidentally checked into the broker a while back.

### CODE REVIEW

**Review by grondo — APPROVED** [review]
Seems like an appropriate change to me. My only thought is if `validate_policy()` really belongs in libjob, though I'm not sure where else it could conveniently be placed (I would say libutil, but I guess that doesn't seem appropriate either, but would at least keep the code from being incorporated into libflux-core.so)

File: src/common/libjob/policy.c [hunk 1]
```diff
@@ -0,0 +1,360 @@
+/************************************************************\
+ * Copyright 2020 Lawrence Livermore National Security, LLC
```
Comment (grondo) [inline src/common/libjob/policy.c:2]: Also, I'm assuming you kept 2020 copyright date here on purpose due to previous copyright date of code (Note: `git blame` shows the actual date of inclusion of the moved code is 2022, but I doubt this matters much!)

Comment (garlick) [reply src/common/libjob/policy.c:2]: Since you took the trouble to look up the actual date the code was created, I'll change it to 2022 - thanks!

**Comment by garlick** [comment]
> My only thought is if `validate_policy()` really belongs in libjob, though I'm not sure where else it could conveniently be placed (I would say libutil, but I guess that doesn't seem appropriate either, but would at least keep the code from being incorporated into libflux-core.so)

Well that's a good point.  Hmm, there's `libfluxutil` which was added by 6e26e00af0224bc7b8e4cf3c58e87e6de8a35b33 in which I said
> We'll create this local library as a catch-all for reusable bits that depend on libflux-core.so.

**Comment by garlick** [comment]
Fixed those things and repushed.  I'll set MWP - thanks!

**Pull request merged**